include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -249,6 +263,44 @@ class fm_index
         // index.m_sigma = largest_char;
     }
 
+    //!\overload
+    void construct(text_t const & text)
+        //!\cond
+        requires is_collection
+        //!\endcond
+    {
+        // text must not be empty
+        if (text.begin() == text.end())
+            throw std::invalid_argument("The text that is indexed cannot be empty.");
+        this->text = &text;
+
+        concatenated_sequences<value_type_t<text_t>> cvec{text};
+        // size + 1: for the rank query we need to go one to the right: rank=#1 BEFORE (i.e. excluding) the position
+        sdsl::bit_vector pos(cvec.concat_size()+1, 0);
+        size_t prefix_sum{0};
+        for (auto && x : cvec)
+        {
+            pos[prefix_sum] = 1;
+            prefix_sum += x.size();
+        }
```

Comment:
Doesn't the `concatenated_sequences` already do this?!

Reply:
Does it? In the docs, I only see `size` for the number of elements in the collection and `concat_size` for the cumulative sum.
But I need the individual sizes of the sequences.

Reply:
It is accessible via `.data()` but it shouldn't be so don't rely on that for now. Using view::join is the generic approach.